### PR TITLE
Change parser and serialize to object directly

### PR DESCRIPTION
**Pull request by hans00**
- Use newer parser [promql-parser](https://github.com/GreptimeTeam/promql-parser)
    - For [this bug](https://github.com/softdevteam/grmtools/issues/124), we need build WASM on i386 machine/docker
    - Setup GitHub Action to build it
- Write-up tests
- Use data conversion, instead of string replace
- Update test script

### CODE REVIEW

**Comment by lmangani** [comment]
Thanks @hans00 for this fantastic contribution!

**Comment by lmangani** [comment]
@hans00 I added npm publish to the action and the new version should be live. Could you confirm it works for you?

**Comment by hans00** [comment]
> @hans00 I added npm publish to the action and the new version should be live. Could you confirm it works for you?

Seems publish on wrong path.

```
❯ tree node_modules/@qxip/promql-parser-js/
node_modules/@qxip/promql-parser-js/
├── build.sh
├── Cargo.lock
├── Cargo.toml
├── js
│   ├── index.js
│   ├── index.test.js
│   └── __snapshots__
│       └── index.test.js.snap
├── LICENSE
├── node_modules
├── package.json
├── pkg
│   └── README.md
├── README.md
└── src
    └── lib.rs
```

https://github.com/lmangani/rust-promql-parser-js/compare/main...hans00:rust-promql-parser-js:patch-1

**Comment by lmangani** [comment]
I see the issue. when building this way, the package version uses the cargo version rather than the package.json version

File: src/lib.rs [hunk 1]
```diff
@@ -1,37 +1,306 @@
...
+impl ToSerde for LabelModifier {
+    fn to_serde(&self) -> Value {
+        match self {
+            LabelModifier::Include(labels) =>
+                json!({ "include": labels.to_serde() }),
+            LabelModifier::Exclude(labels) =>
+                json!({ "exclude": labels.to_serde() }),
+        }
+    }
+}
```
Comment (yuanbohan) [inline src/lib.rs:146]: BTW, Include/Exclude matches:
- on/ignoring for BinaryExpr matching
- by/without for Aggregation

**Pull request merged**